pkg/appmanager/resourceConfig.go

```diff
@@ -836,6 +838,234 @@ func setSourceAddrTranslation(snatPoolName string) SourceAddrTranslation {
+		Name:      "0",
+		HttpReply: true,
+		Location:  value,
+		Redirect:  true,
+		Request:   true,
+	}
+	redirectGUID := uuid.New()
+	rules = append(rules, &Rule{
+		Name:       fmt.Sprintf("app-root-redirect-rule-%s", redirectGUID.String()),
+		Actions:    []*action{redirectAction},
+		Conditions: redirectConditions,
+	})
+
+	pathCondition := &condition{
+		Name:            "0",
+		CaseInsensitive: true,
+		Equals:          true,
+		External:        true,
+		HTTPURI:         true,
+		Index:           0,
+		Path:            true,
+		Present:         true,
+		Remote:          true,
+		Request:         true,
+		Values:          []string{value},
+	}
+	if target != "" {
+		forwardConditions = append(forwardConditions, &condition{
+			Name:            "0",
+			CaseInsensitive: true,
+			Equals:          true,
+			External:        true,
+			Host:            true,
+			HTTPURI:         true,
+			Index:           0,
+			Present:         true,
+			Remote:          true,
+			Request:         true,
+			Values:          []string{target},
+		})
+		pathCondition.Name = "1"
+	}
+	forwardConditions = append(forwardConditions, pathCondition)
+	forwardAction := &action{
+		Forward: true,
+		Name:    "0",
+		Pool:    poolName,
+		Request: true,
+	}
+	forwardGUID := uuid.New()
+	rules = append(rules, &Rule{
+		Name:       fmt.Sprintf("app-root-forward-rule-%s", forwardGUID.String()),
+		Actions:    []*action{forwardAction},
+		Conditions: forwardConditions,
+	})
+
+	return rules
+}
+
+func processURLRewrite(target, value string) *Rule {
+	var targetURL *url.URL
+	var actions []*action
+	var conditions []*condition
+	var err error
+
+	if target != "" {
+		if !(strings.HasPrefix(target, "http://") || strings.HasPrefix(target, "https://")) {
+			target = "http://" + target
+		}
+
+		targetURL, err = url.Parse(target)
+		if err != nil {
+			log.Errorf("Error parsing url-rewrite target url: %s, Error: %v, skipping", target, err)
+			return nil
+		}
+
+		conditionName := 0
+		if targetURL.Host != "" {
+			conditions = append(conditions, &condition{
+				Name:            fmt.Sprintf("%d", conditionName),
+				CaseInsensitive: true,
+				Equals:          true,
+				External:        true,
+				Host:            true,
+				HTTPURI:         true,
+				Index:           0,
+				Present:         true,
+				Remote:          true,
+				Request:         true,
+				Values:          []string{targetURL.Host},
+			})
+			conditionName++
+		}
+		if targetURL.Path != "" {
+			conditions = append(conditions, &condition{
+				Name:            fmt.Sprintf("%d", conditionName),
+				CaseInsensitive: true,
+				Equals:          true,
+				External:        true,
+				HTTPURI:         true,
+				Index:           0,
+				Path:            true,
+				Present:         true,
+				Remote:          true,
+				Request:         true,
+				Values:          []string{targetURL.Path},
+			})
+		}
+	}
+
+	if !(strings.HasPrefix(value, "http://") || strings.HasPrefix(value, "https://")) {
```

Comment:
Checkou routing.go to possibly combine conditions

Reply:
looked at routing.go it uses multiple conditions, but I need to make path segments resolvable for these annotation rules.

Reply:
Fixed.